framework/src/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -809,7 +820,13 @@ public S leastUpperBound(S other) {
         {
             V otherVal = other.thisValue;
             V myVal = thisValue;
-            V mergedVal = myVal == null ? null : myVal.leastUpperBound(otherVal);
+            V mergedVal =
```

Comment:
I don't really know.  I would guess it's because the other the myVal equivalent is from a map, so the map must have the property that values are never null.